app/src/main/java/com/insudev/euvictodo/buisnessLogic/UseCase.kt

```diff
@@ -0,0 +1,115 @@
+        .build()
+
+    private val service = retrofit.create<TodoService>(TodoService::class.java)
+
+    private val api = TodoRetroService(service)
+
+    companion object {
+        val instance = UseCase()
+    }
+
+    fun addTodo(content: String, prevId: Int): Observable<TodoListChangeResult> {
+        return Observable.just(
+            TodoListChangeResult.Completed(
+                TodoModel(
+                    prevId, System.currentTimeMillis() / 1000, content,
+                    false, arrayListOf("")
+                ), null
+            ) as TodoListChangeResult
+        )
+            .startWith(TodoListChangeResult.Pending())
+            .onErrorReturnItem(TodoListChangeResult.Error("Błąd dodawnaia notatki"))
+    }
+
+    fun sync(toSync: ArrayList<TodoModel>): Observable<TodoListChangeResult> {
+        return api.sync(toSync)
+            .map {
+                if (it) TodoListChangeResult.Completed("Zsynchronizowano", null)
+                else TodoListChangeResult.Error("Błąd synchronizacji")
+            }.startWith(TodoListChangeResult.Pending())
+            .onErrorReturnItem(TodoListChangeResult.Error("Błąd synchronizacji"))
+    }
+
+    fun getFilteredTodos(
+        filter: Filters,
+        sorting: Sorting,
+        skip: Int,
+        take: Int
+    ): Observable<TodoListChangeResult> {
+        return api.getFilteredTodos(filter, sorting, skip, take).map {
+            if (it != null) {
+                if (it.size == 0) {
+
+                    TodoListChangeResult.Error("Nie ma wiecej notatek")
+                } else
+                    TodoListChangeResult.Completed(filter, it)
+            } else TodoListChangeResult.Error("")
+        }
+            .startWith(TodoListChangeResult.Pending())
+            .onErrorReturnItem(TodoListChangeResult.Error("Error"))
+
+
+    }
+
+    fun getEmpty(): Observable<TodoListChangeResult> {
+        return Observable.fromArray(ArrayList<TodoModel>()).map {
+            if (it != null) {
+
+                TodoListChangeResult.Completed(null, it)
+            } else TodoListChangeResult.Error("")
+        }
+            .startWith(TodoListChangeResult.Pending())
+            .onErrorReturnItem(TodoListChangeResult.Error("Error"))
+
+
+    }
+
+    fun getNextId(): Observable<Int> {
+        return api.getNextId()
+    }
+
+    fun updateStatus(
+        id: Int,
+        todoList: ArrayList<ModelInterface>,
+        syncList: ArrayList<TodoModel>
+    ): Observable<TodoListChangeResult> {
+        val todos = ArrayList(todoList.filterIsInstance<TodoModel>())
+        var toSync: TodoModel
+        toSync = try {
+            val index = todos.indexOfFirst { x -> x.id == id }
+            todos.removeAt(index)
+        } catch (e: ArrayIndexOutOfBoundsException) {
+
+            val index = syncList.indexOfFirst { x -> x.id == id }
+            syncList.removeAt(index)
+        }
+        toSync.status = !toSync.status
+        return Observable.just(
```

Comment:
powinniśmy użyć Observable.create lub coś podobnego, ze względu na to, że obcinamy całą asynchroniczność w tym przypadku